tdms/cmake/targets.cmake

```diff
@@ -44,6 +45,7 @@ function(test_target)
             ${Matlab_MX_LIBRARY}
             ${Matlab_MAT_LIBRARY}
             ${LIBCXX_LIBRARY}
+            ${HDF5_CXX_LIBRARIES}
```

Comment:
> * the target tdms isn't told to explicitly link to the HDF5 library found during configuration, so the linker just picks up the first one it finds. Adding
>
>         target_link_libraries(tdms ${HDF5_CXX_LIBRARIES})
>   should do the trick

Mosè's fix is also needed for the dynlib target (at least on my ARM64 Mac which tries to link the wrong HDF5 libraries).